lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -351,19 +364,28 @@ def _get_requirer_app_data(self, relation: Relation) -> RequirerAppData:
             raise NotReadyError(relation)
 
         databag = relation.data[app]
-        remote_app_data: Dict[str, Union[int, str]] = {}
+        remote_app_data: Dict[str, Any] = {}
 
-        for k in ("model", "name", "mode", "strip-prefix", "redirect-https"):
+        for k in ("model", "name", "strip-prefix", "redirect-https", "scheme"):
             v = databag.get(k)
             if v is not None:
                 remote_app_data[k] = v
 
         _validate_data(remote_app_data, INGRESS_REQUIRES_APP_SCHEMA)
 
-        remote_app_data["strip-prefix"] = bool(remote_app_data.get("strip-prefix", False))
-        remote_app_data["redirect-https"] = bool(
-            remote_app_data.get("redirect-https", "false") == "true"
-        )
+        def _bool(val):
+            return val == "true"
+
+        def _cast(key, type_, default):
+            raw = remote_app_data.get(key)
+            if not raw:
+                return default
+            return type_(raw)
+
+        # deserialize some strings to more pythonic types
+        remote_app_data["scheme"] = _cast("scheme", Scheme, Scheme.HTTP)
```

Comment:
The `_cast` could raise. I think it would be simpler to add scheme validation to `_validate_data()` and omit this line.